pages/_app.js

```diff
@@ -17,6 +23,33 @@ export default class MyApp extends App {
 		return { pageProps }
 	}
 
+	componentDidUpdate() {
```

Comment:
If the current route in state does not match the route when this component rerenders, update the state and track the page.